### PR TITLE
partial test and code format during pkg test

### PR DESCRIPTION
**Pull request by johnnychen94**
- add code format in pkg test. The code format will only run incrementally based on `git diff`, so it's much faster
- do an incremental test on problems locally (also based on `git diff`). CI still tests against all problems.
- run each problem in a sandbox module so it won't get affected by other codes.
- remove auto format CI since the format is run locally during pkg test

I set `git diff HEAD~1` because it might be possible that someone commits before he does `pkg> test`.

preview:

```text
┌ Info: testing 1 problems
└   CI = false
Test Summary: | Pass  Total
LeetCode.jl   |    5      5
Formatting /Users/jc/Documents/Julia/LeetCode.jl/src/problems/1659.minimum-operations-to-reduce-x-to-zero.jl
Formatting /Users/jc/Documents/Julia/LeetCode.jl/test/problems/1659.minimum-operations-to-reduce-x-to-zero.jl
    Testing LeetCode tests passed 

```

```text
┌ Info: testing 43 problems
└   CI = true
Test Summary: | Pass  Total
LeetCode.jl   |  122    122
Formatting /Users/jc/Documents/Julia/LeetCode.jl/test/problems/problems.jl
    Testing LeetCode tests passed 
```

### CODE REVIEW

**Comment by johnnychen94** [comment]
Why isn't there an UnitTest CI?

**Comment by findmyway** [comment]
https://travis-ci.com/github/JuliaCN/LeetCode.jl

> Builds have been temporarily disabled for public repositories due to a negative credit balance. 

https://discourse.julialang.org/t/easy-workflow-file-for-setting-up-github-actions-ci-for-your-julia-package/49765

**Comment by johnnychen94** [comment]
Okay, I see, should I set up one?

**Comment by findmyway** [comment]
Yeah, that would be great.

File: Project.toml [hunk 1]
```diff
@@ -7,11 +7,14 @@ version = "0.1.0"
 BenchmarkTools = "6e4b80f9-dd63-53aa-95a3-0cdb28fa8baf"
 DataStructures = "864edb3b-99cc-5e75-8d2d-829cb0a9cfe8"
 Dates = "ade2ca70-3891-5945-98fb-dc099432e06a"
+JuliaFormatter = "98e50ef6-434e-11e9-1051-2b60c6c9e899"
+LibGit2 = "76f85450-5226-5b5a-8eaa-529ad045b433"
```
Comment (findmyway) [inline Project.toml:11]: I think these two packages are for test only? If so, we can create a new `Project.toml` file under `test` folder.

**Review by findmyway — APPROVED** [review]

**Comment by johnnychen94** [comment]
One thing that might worth mentioning, only staged problems will be tracked by current `diff_files`, tests won't run `89.xxx.jl` file in the following case:

image


I'm not an expert on git and I don't know how to fix it 😨  I don't know if this needs a fix, though.

**Pull request merged**